contrib/docker-scripts/download_binaries.py

```diff
@@ -0,0 +1,125 @@
+import re
+import requests
+import os
+import json
+import logging
+import sys
+import shutil
+
+
+# Logger class for easier logging setup
+class Logger:
+    def __init__(self):
+        self.log = logging.getLogger()
+        self.log.setLevel(logging.INFO)
+        self.handler = logging.StreamHandler(sys.stdout)
+        self.handler.setLevel(logging.DEBUG)
+        self.formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+        self.handler.setFormatter(self.formatter)
+        self.log.addHandler(self.handler)
+
+
+# Initialize logger instance
+logger = Logger()
+
+# Define the path where upgrades will be stored, using an environment variable for the base path
+upgrade_path = f'{os.environ["DAEMON_HOME"]}/cosmovisor/upgrades/'
+
+
+# Function to find the latest patch version of a binary based on major and optional minor version
+def find_latest_patch_version(major_version, minor_version=None):
+    # Define a regex pattern to match version directories
+    pattern = re.compile(
+        f"v{major_version}\.{minor_version}\.(\d+)" if minor_version else f"v{major_version}\.0\.(\d+)")
+    # List directories that match the version pattern
+    versions = [folder for folder in os.listdir(upgrade_path) if pattern.match(folder)]
+    if versions:
+        try:
+            # Find the maximum version, assuming it's the latest patch
+            latest_patch_version = max(versions)
+            # Return the path to the binary of the latest patch version
+            return os.path.join(upgrade_path, latest_patch_version, "bin", "zetacored")
+        except ValueError as e:
+            logger.log.error(f"Error finding latest patch version: {e}")
+            return None
+    return None
+
+
+# Function to replace an old binary with a new one
+def replace_binary(source, target):
+    try:
+        # Log deletion of old binary
+        if os.path.exists(target):
+            logger.log.info(f"Deleted old binary: {target}")
+            os.remove(target)
+        # Copy the new binary to the target location
+        shutil.copy(source, target)
+        logger.log.info(f"Binary replaced: {target} -> {source}")
+    except Exception as e:
+        logger.log.error(f"Error replacing binary: {e}")
+
+
+# Parse JSON from an environment variable to get binary download information
+info = json.loads(os.environ["DOWNLOAD_BINARIES"])
+
+try:
+    # Iterate over binaries to download
+    for binary in info["binaries"]:
+        download_link = binary
+        # Log download link
+        logger.log.info(f"DOWNLOAD LINK: {download_link}")
+        split_download_link = download_link.split("/")
+        # Log split download link parts
```

Comment:
this script needs to be updated to work with the way we are managing the upgrade handler version